### PR TITLE
DIFI-382: run `collect` on every page change

### PR DESCRIPTION
**Pull request by EdPutans**

### CODE REVIEW

**Review by Luciam91 — APPROVED** [review]

File: src/context/CollectorContext.tsx [hunk 1]
```diff
@@ -260,150 +261,168 @@ export function CollectorProvider({
...
     handlers,
-    initialDelay,
     getIdleStatusDelay,
     setIdleTimeout,
-    log,
     trackEvent,
     addPageTriggers
   ])
 
+  // temp hack for collector to onl fire once.
```
Comment (AdenFraser) [inline src/context/CollectorContext.tsx:386]: Can we remove commented code please?

**Review by dbacall — APPROVED** [review]
Can this be used to prevent the exit intent popping up on booking when you select a location?

**Review by AdenFraser — CHANGES_REQUESTED** [review]
Can we remove commented code?

File: src/hooks/useRunOnPathChange.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  skip?: boolean
+  delay?: number
+}
+// takes a function and runs it when the path changes
+// optionally takes a skip condition and init delay
+const useRunOnPathChange = (func: FuncProp, config?: Config) => {
+  const [lastCollectedPath, setLastCollectedPath] = useState<string>('')
+  const { log } = useLogging()
+
+  useEffect(() => {
```
Comment (AdenFraser) [inline src/hooks/useRunOnPathChange.ts:15]: This is not a very nice pattern. Why are we not using window event listeners?

Comment (EdPutans) [reply src/hooks/useRunOnPathChange.ts:15]: I've tried using window.location and the history API, but not all page changes get picked up with that - booking flow is an example

File: src/context/CollectorContext.tsx (hunk 1 above)
Comment (EdPutans) [reply src/context/CollectorContext.tsx:386]: #oops

File: src/hooks/useRunOnPathChange.ts (hunk 2 above)
Comment (AdenFraser) [reply src/hooks/useRunOnPathChange.ts:15]: Accessing them directly on global window, or by adding event listener (for location and/or popstate)?

Comment (EdPutans) [reply src/hooks/useRunOnPathChange.ts:15]: global window. 
https://stackoverflow.com/questions/3522090/event-when-window-location-href-changes guessing you suggest something like this? Just found it

Comment (EdPutans) [reply src/hooks/useRunOnPathChange.ts:15]: Today i learned.
https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event

Will try again 👍

**Review by AdenFraser — APPROVED** [review]

**Pull request merged**